### ISSUE TITLE
rclone config for crypt does not allow editing of show_mapping value

### ISSUE THREAD

**Issue by andrewrech** [rclone/rclone#3416]
#### What is the problem you are having with rclone?

`rclone config` crypt menu option to edit "show_mapping" does not work

#### What is your rclone version (output from `rclone version`)

```
rclone v1.48.0-DEV
- os/arch: linux/amd64
- go version: go1.12.1
```

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)

Linux yidhra 4.15.0-1044-aws #46-Ubuntu SMP Thu Jul 4 13:38:28 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

####  Which cloud storage system are you using? (eg Google Drive)

crypt

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)

rclone config -> 

```
Value "show_mapping" = "true"
Edit? (y/n)>
y) Yes
n) No
y/n> y
--------------------
[crypt]
type = <redacted>
remote = <redacted>
filename_encryption = standard
directory_name_encryption = true
password = *** ENCRYPTED ***
password2 = *** ENCRYPTED ***
show_mapping = true
--------------------
```

After saying "Yes I want to edit", rclone returns to the main prompt rather than allowing one to edit the `show_mapping` value.

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)

N/A

**Comment by ncw** [comment]
Hmm, the show_mapping config item is defined as hidden in the configurator as it is not normally the sort of thing you want configured permanently, you just use `--crypt-show-mapping`.

https://github.com/rclone/rclone/blob/99eaa76dc80e310aa1cea7bcec1de742e61de339/backend/crypt/crypt.go#L71-L83

So I think it shouldn't have shown the `show_mapping` at all in the configurator but it returning is probably due to the fact it is marked as hidden.

What do you think the best solution for this is?
- don't show the hidden values at all?
- allow editing of the hidden values if they are set?
- something else?

**Comment by andrewrech** [comment]
Thanks for your response.

I agree, at least for my use case, that there is little reason to have it on permanently and showing it only if set may result in confusion, so I would vote for option 1 - don't show the hidden values at all.

**Comment by ncw** [comment]
Ok here is a go at option 1 - can you give it a test and see what you think?

https://beta.rclone.org/branch/v1.48.0-153-g8e0756d5-fix-3416-config-beta/ (uploaded in 15-30 mins)

**Comment by andrewrech** [comment]
Nice - no longer prompts for editing `show_mapping`.

Minor note is that there is still a prompt for advanced config when there are no advanced config options that are editable. This shows the sequence:

```
Edit advanced config? (y/n) 
y) Yes
n) No
y/n> y
--------------------
[crypt]
type = crypt
remote = y
filename_encryption = y
directory_name_encryption = true
password = *** ENCRYPTED ***
password2 = *** ENCRYPTED ***
show_mapping = true
--------------------
y) Yes this is OK
e) Edit this remote
d) Delete this remote
```

**Comment by ncw** [comment]
Ah well spotted!

Try this

https://beta.rclone.org/branch/v1.48.0-166-gdf326eee-fix-3416-config-beta/ (uploaded in 15-30 mins)

**Comment by andrewrech** [comment]
Perfect!

**Comment by ncw** [comment]
I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.49
